syntheseus/reaction_prediction/utils/inference.py

```diff
@@ -1,32 +1,61 @@
 from pathlib import Path
 from typing import Any, Dict, List, Sequence, Union
 
-from syntheseus.interface.models import Prediction
+from syntheseus.interface.bag import Bag
+from syntheseus.interface.molecule import Molecule
+from syntheseus.interface.reaction import MultiProductReaction, SingleProductReaction
 from syntheseus.reaction_prediction.chem.utils import molecule_bag_from_smiles
 
 
-def process_raw_smiles_outputs(
-    input: Any, output_list: List[str], kwargs_list: List[Dict[str, Any]]
-) -> Sequence[Prediction]:
-    """Convert raw SMILES outputs into a list of `Prediction` objects.
+def process_raw_smiles_outputs_backwards(
+    input: Molecule, output_list: List[str], kwargs_list: List[Dict[str, Any]]
+) -> Sequence[SingleProductReaction]:
```

Comment:
Yes, all the uses just supply a single dictionary with the `metadata` key. We could refactor it, or potentially leave it as is in case we want to supply an `identifier` kwarg? (see above for this)

Reply:
I would maybe slightly lean towards refactoring to assume the kwargs are really things to put into the metadata, instead of having extra complexity anticipating something we don't have immediate plans on adding. But no strong opinion, either way is fine.

Reply:
Changed this too.